EventFilter/CTPPSRawToDigi/plugins/CTPPSPixelDigiToRaw.h

```diff
@@ -0,0 +1,55 @@
+#ifndef CTPPSPixelDigiToRaw_H
+#define CTPPSPixelDigiToRaw_H
+
+/** \class CTPPSPixelDigiToRaw_H
+ */
+
+#include "FWCore/Framework/interface/ESWatcher.h"
+#include "FWCore/Framework/interface/EDProducer.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "CondFormats/DataRecord/interface/CTPPSPixelDAQMappingRcd.h"
+#include "CondFormats/CTPPSReadoutObjects/interface/CTPPSPixelDAQMapping.h"
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/CTPPSDigi/interface/CTPPSPixelDigi.h"
+#include "CondFormats/CTPPSReadoutObjects/interface/CTPPSPixelFramePosition.h"
+
+
+class CTPPSPixelDigiToRaw final : public edm::EDProducer {
```

Comment:
legacy module base is not allowed for new modules.
Please use one of the MT-aware modules
https://twiki.cern.ch/twiki/bin/view/CMSPublic/FWMultithreadedFrameworkModuleTypes